### PR TITLE
Release/1.9.0

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: includes/compatibility.php [hunk 1]
```diff
@@ -44,10 +50,73 @@ public static function register_actions() {
...
+
+				var url = './edit.php?post_type=<?php echo $typenow; // XSS ok. ?>&action=elementor_new_post';
+
+				dropdown.insertAdjacentHTML( 'afterbegin', '<a href="' + url + '">Elementor</a>' );
+			} );
+		</script>
+		<?php
+	}
+
+	public static function admin_new_post() {
```
Comment (KingYes) [inline includes/compatibility.php]: move

File: includes/compatibility.php [hunk 2]
```diff
@@ -44,10 +50,73 @@ public static function register_actions() {
...
+
+		if ( ! User::is_current_user_can_create_new_post( $post_type ) ) {
+			return;
+		}
+
+		$post_id = wp_insert_post( [
+			'post_type' => $post_type,
+		] );
+
+		add_post_meta( $post_id, '_elementor_edit_mode', 'builder' );
```
Comment (KingYes) [inline includes/compatibility.php]: remove

File: includes/editor-templates/panel.php [hunk 3]
```diff
@@ -1,9 +1,14 @@
 <?php
 namespace Elementor;
 
+
```
Comment (KingYes) [inline includes/editor-templates/panel.php]: remove

File: includes/managers/elements.php [hunk 4]
```diff
@@ -156,6 +157,31 @@ public function render_elements_content() {
...
+		if ( $autosave ) {
+			$deleted = wp_delete_post_revision( $autosave->ID );
+			$success = $deleted && ! is_wp_error( $deleted );
+		} else {
+			$success = true;
+		}
+
+		if ( $success ) {
+			wp_send_json_success();
+		} else {
```
Comment (KingYes) [inline includes/managers/elements.php]: remove

File: includes/user.php [hunk 5]
```diff
@@ -88,6 +88,26 @@ public static function is_current_user_can_edit( $post_id = 0 ) {
 		return true;
 	}
 
+	public static function is_current_user_can_create_new_post( $post_type ) {
```
Comment (KingYes) [inline includes/user.php]: rename

File: includes/utils.php [hunk 6]
```diff
@@ -144,6 +144,45 @@ public static function get_preview_url( $post_id ) {
...
+		 * @param int    $post_id     Post ID.
+		 */
+		$wp_preview_url = apply_filters( 'elementor/utils/wp_preview_url', $wp_preview_url, $post_id );
+
+		return $wp_preview_url;
+	}
+
+
+	public static function get_exit_to_dashboard_url( $post_id ) {
+		$exit_url = get_edit_post_link( $post_id, '' );
```
Comment (KingYes) [inline includes/utils.php]: raw

**Review by KingYes — CHANGES_REQUESTED** [review]

File: includes/compatibility.php (hunk 1 above)
Comment (matipojo) [reply includes/compatibility.php]: 👍

File: includes/compatibility.php (hunk 2 above)
Comment (matipojo) [reply includes/compatibility.php]: 👍

File: includes/editor-templates/panel.php (hunk 3 above)
Comment (matipojo) [reply includes/editor-templates/panel.php]: 👍

File: includes/managers/elements.php (hunk 4 above)
Comment (matipojo) [reply includes/managers/elements.php]: 👍

File: includes/user.php (hunk 5 above)
Comment (matipojo) [reply includes/user.php]: 👍

File: includes/utils.php (hunk 6 above)
Comment (matipojo) [reply includes/utils.php]: 👍

**Pull request merged**